packages/openid4vc/src/openid4vc-holder/OpenId4VciHolderService.ts

```diff
@@ -367,6 +368,55 @@ export class OpenId4VciHolderService {
     }
   }
 
+  public async refreshAccessToken(agentContext: AgentContext, options: OpenId4VciTokenRefreshOptions) {
+    const oauth2Client = this.getOauth2Client(agentContext)
+
+    const authorizationServerMetadata = getAuthorizationServerMetadataFromList(
+      options.issuerMetadata.authorizationServers,
+      options.issuerMetadata.authorizationServers.find(
+        (server) =>
+          // TODO: add type in @openid4vc/oauth2
+          Array.isArray(server.grant_types_supported) && server.grant_types_supported.includes('refresh_token')
+      )?.issuer ?? options.issuerMetadata.authorizationServers[0].issuer
```

Comment:
Multiple authorization servers could support it. We need to check based on the current access token which server to use.

Not sure if we can know it from the token, in which case we will have to store it alongside the refresh token. As this can lead to weird errors

Reply:
> We need to check based on the current access token which server to use.

Do you mean whether the pre-authorization server or the other one should be used?

If we need to have the possibility of both methods, we'll need to actually have multiple refresh tokens as a JWT, with enough information to connect it back to a session **and** the correct authentication method.